### PR TITLE
AuthnContext comparison type is mandatory for AuthnContextClassRef

### PR DESCRIPTION
**Pull request by fmartelli**
This PR introduce a new configuration parameter for AuthnContext comparison type.
This parameter will be used set comparison attribute of the given AuthnContextClassRef. In case of missing value, "exact" comparison type is used.

### CODE REVIEW

**Comment by mmoayyed** [comment]
Please document the setting, both in the project documentation and in code.

File: core/cas-server-core-configuration/src/main/java/org/apereo/cas/configuration/model/support/pac4j/Pac4jProperties.java [hunk 1]
```diff
@@ -448,6 +448,7 @@ public void setSecret(final String secret) {
         private String clientName;
         private boolean forceAuth;
         private String authnContextClassRef;
+        private String authnContextComparisonType;
```
Comment (mmoayyed) [inline core/cas-server-core-configuration/src/main/java/org/apereo/cas/configuration/model/support/pac4j/Pac4jProperties.java]: Please default the comparison type to be EXACT.

File: support/cas-server-support-pac4j/src/main/java/org/apereo/cas/support/pac4j/config/support/authentication/Pac4jAuthenticationEventExecutionPlanConfiguration.java [hunk 2]
```diff
@@ -238,6 +240,11 @@ private void configureSamlClient(final Collection<BaseClient> properties) {
                     cfg.setForceAuth(saml.isForceAuth());
 
                     if (StringUtils.isNotBlank(saml.getAuthnContextClassRef())) {
+                        final String authContextComparisonType = StringUtils.isNotBlank(saml.
+                                getAuthnContextComparisonType())
+                                        ? AuthnContextComparisonTypeEnumeration.EXACT.toString()
```
Comment (mmoayyed) [inline support/cas-server-support-pac4j/src/main/java/org/apereo/cas/support/pac4j/config/support/authentication/Pac4jAuthenticationEventExecutionPlanConfiguration.java]: Simply just use `EXACT`. No need to import opensaml explicitly?

**Comment by fmartelli** [comment]
Hi Misagh, I think I have addressed your comments (not sure about the documentation).
Please, review and let me know.
Regards

**Comment by mmoayyed** [comment]
Thank you very much for the contribution. Outstanding work. Keep 'em coming!

**Pull request merged**